packages/runtime-node/src/worker-thread-host.ts

```diff
@@ -0,0 +1,33 @@
+import { MessagePort, Worker } from 'node:worker_threads';
+
+import { BaseHost, Message } from '@wixc3/engine-core';
+
+export class WorkerThreadHost extends BaseHost {
+    constructor(private host: Worker | MessagePort) {
+        super();
+        host.on('message', this.onMessage);
+    }
+
+    public addEventListener(type: 'message', handler: (event: { data: any }) => void): void {
+        const handlers = this.handlers.get(type);
+        if (!handlers) {
+            this.handlers.set(type, new Set([handler]));
```

Comment:
a nice future refactor is to use SetMultiMap for handlers in BaseHost to avoid managing it in every extending host. not for this PR though